filesystem/filesystem_test.go

```diff
@@ -0,0 +1,72 @@
+package filesystem
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func MockSlowGetFileSystemInfo() (interface{}, error) {
+	/* Run a command that will definitely time out */
+	cmd := exec.Command("sleep", "5")
+
+	outCh := make(chan []byte, 1)
+	errCh := make(chan error, 1)
+
+	var out interface{}
+	var err error
+
+	go func() {
+		_out, _err := cmd.Output()
+		if _err != nil {
+			errCh <- fmt.Errorf("df failed to collect filesystem data: %s", _err)
+			return
+		}
+		outCh <- _out
+	}()
+
+WAIT:
+	for {
+		select {
+		case res := <-outCh:
+			if res != nil {
+				out, err = parseDfOutput(string(res))
+			} else {
+				out, err = nil, fmt.Errorf("df failed to collect filesystem data")
+			}
+			break WAIT
+		case err = <-errCh:
+			out = nil
+			break WAIT
+		case <-time.After(2 * time.Second):
+			// Kill the process if it takes too long
+			if killErr := cmd.Process.Kill(); killErr != nil {
+				log.Fatal("failed to kill:", killErr)
```

Comment:
I thought it made sense to ensure that multiple `df` calls spawned by gohai weren't running at the same time. The change was prompted by a case where `df` was taking long and successive gohai calls kept spawning new, time-intensive processes. This will kill `df` before it returns, and not interrupt the main `gohai` process